### DISCUSSION TITLE
Incorrect X-XSS-Protection setting for "Caddy 2.x with Cloudflare DNS" in the wiki?

### DISCUSSION THREAD

**Discussion by KelvinHu** [dani-garcia/vaultwarden#4009] [Q&A] [answered]
Hi all,

Mozilla's MDN defines "0" as disabling X-XSS-Protection [1], but this is incorrectly commented as enabling XSS Protection. If the intention was to enable XSS Protection then it should be "1;" and "1; mode=block" if we are blocking as suggested by the comment.

Was this setting made in error?

[1] https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

**Comment by BlackDex** [comment]
If you read that page correctly it states that this header isn't used and overridden by CSP which we use.

**Comment by stefan0xC** [comment, marked as answer]
According to the [revision history of the wiki page](https://github.com/dani-garcia/vaultwarden/wiki/Caddy-2.x-with-Cloudflare-DNS/_history) (which anyone can edit) it was a conscious decision. It also links to the page you have mentioned. So the only thing that's wrong might be the comment that has been left.

**Reply by dadatuputi** [reply]
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection, it is not recommended for modern browsers, however the default value is usually `X-XSS-Protection: 1`. Explicitly setting it to 0 is a bit confusing without context (and why I'm here). 

Wouldn't it be better to remove it entirely from the wiki proxy Caddyfile, which should leave modern browsers disabling it and old browsers using it if necessary? 

Otherwise, I'll improve the comment above the line in the wiki page, because it reads as an INSECURE setting, not one that improves the security of Caddy

**Reply by stefan0xC** [reply]
According to the section [Vulnerabilities caused by XSS filtering](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection#vulnerabilities_caused_by_xss_filtering) it's recommended to completely disable XSS-Filtering by setting it to 0.
